baseplate/integration/pyramid.py

```diff
@@ -123,9 +153,14 @@ def _on_new_request(self, event):
             return
 
         trace_info = None
-        if self.trust_trace_headers:
+        if self.header_trust_handler.should_trust_trace_headers(request):
```

Comment:
probably just call this once and assign it to a variable

Reply:
Re-upping this

Reply:
That won't work for hmac validation since the hmac will be different for each request

Reply:
Not a `self.` variable, just something in here, it'll be different for each request.

```
trust_headers = self.header_trust_handler.should_trust_trace_headers(request)
```

Reply:
oh, got it. Not sure I see the point since it is only used once now that there is should_trust_edge_context_payload